### PR TITLE
Added code to run .py files

### PR DESCRIPTION
**Pull request by Loghijiaha**
## Checklist

- [x] I have referenced the Jira issue related to my changes in one or more commit messages
- [x] Unit tests pass locally with my changes
- [ ] I have added documentation as necessary
- [ ] No Javadoc warnings were introduced with my changes
- [x] No spotbugs warnings were introduced with my changes
- [ ] I have interactively tested my changes

## Types of changes

- [ ] Infrastructure change (non-breaking change which updates dependencies or improves infrastructure)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/ml/IPythonBuilder.java [hunk 1]
```diff
@@ -130,7 +142,9 @@ public String getCode() {
 
     enum FileExtension {
         ipynb,
-        json
+        json,
+        py,
+        txt
```
Comment (shivaylamba) [inline src/main/java/io/jenkins/plugins/ml/IPythonBuilder.java:147]: @Loghijiaha are you also looking at adding support for .csv files?

File: src/main/java/io/jenkins/plugins/ml/IPythonBuilder.java [hunk 2]
```diff
@@ -87,7 +87,16 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath ws, @Nonnull Launc
...
+                    FileExtension ext;
+                    if (extension.equals("ipynb")) {
+                        ext = FileExtension.ipynb;
+                    } else if (extension.equals("json")) {
+                        ext = FileExtension.json;
+                    } else if (extension.equals("py")) {
+                        ext = FileExtension.py;
+                    } else {
+                        ext = FileExtension.txt;
+                    }
```
Comment (kinow) [inline src/main/java/io/jenkins/plugins/ml/IPythonBuilder.java:99]: I think we can replace this `if/elseif/else` by one call to [`.valueOf`](https://docs.oracle.com/javase/7/docs/api/java/lang/Enum.html#valueOf(java.lang.Class,%20java.lang.String)) @Loghijiaha (sorry, don't have the IDE to confirm or share a code snippet). If you want to have a default value, just catch the exception and set ext to text I think. Hope that makes sense.

File: src/main/java/io/jenkins/plugins/ml/IPythonBuilder.java (hunk 1 above)
Comment (Loghijiaha) [reply src/main/java/io/jenkins/plugins/ml/IPythonBuilder.java:147]: Its a good idea @shivaylamba. So far, my concern was about the code and notebook. I will take care of data set files in the next coding phase.

Comment (shivaylamba) [reply src/main/java/io/jenkins/plugins/ml/IPythonBuilder.java:147]: Alright Loghi

File: src/main/java/io/jenkins/plugins/ml/IPythonBuilder.java (hunk 2 above)
Comment (Loghijiaha) [reply src/main/java/io/jenkins/plugins/ml/IPythonBuilder.java:99]: That will keep the code clean. :+1:

File: src/main/java/io/jenkins/plugins/ml/IPythonBuilder.java [hunk 3]
```diff
@@ -110,6 +117,9 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath ws, @Nonnull Launc
                                     }
                             }
                             break;
+                        case py:
+                            listener.getLogger().println(StringUtils.stripStart(interpreterManager.invokeInterpreter(tempFilePath.readToString()), "%text"));
```
Comment (kinow) [inline src/main/java/io/jenkins/plugins/ml/IPythonBuilder.java:121]: I **think** this doesn't work with agents, unless `invokeInterpreter` is doing using a `Callable`. But that's for a future issue I guess. Approving! :rocket:

Comment (Loghijiaha) [reply src/main/java/io/jenkins/plugins/ml/IPythonBuilder.java:121]: I ran with an agent connected in my local machine, it works fine if the file exists in the agent workspace. If it is in the master's workspace, it throws NoSuchFile Error. We can do the issue in our next release :heart:

**Pull request merged**